app/src/main/java/chat/rocket/android/chatroom/adapter/UrlPreviewViewHolder.kt

```diff
@@ -32,8 +32,11 @@ class UrlPreviewViewHolder(itemView: View,
             text_title.content = data.title
             text_description.content = data.description ?: ""
 
+            val tabsbuilder = CustomTabsIntent.Builder()
+            val customTabsIntent = tabsbuilder.build()
+
             url_preview_layout.setOnClickListener { view ->
-                view.context.startActivity(Intent(Intent.ACTION_VIEW, Uri.parse(data.rawData.url)))
+                customTabsIntent.launchUrl(context, Uri.parse(data.rawData.url.toLowerCase()))
```

Comment:
You can use `data.rawData.url.toLowerCase().toUri`.

Fresh from Android KTX -> https://android.github.io/android-ktx/core-ktx/androidx.net/kotlin.-string/to-uri.html.